rust/automerge/src/op_observer/vec_observer.rs

```diff
@@ -0,0 +1,561 @@
+#![allow(dead_code)]
+
+use core::fmt::Debug;
+
+use crate::{marks::Mark, ObjId, OpObserver, Prop, ReadDoc, ScalarValue, Value};
+
+use crate::sequence_tree::SequenceTree;
+
+use crate::op_observer::BranchableObserver;
+use crate::op_observer::{Patch, PatchAction};
+
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub enum TextRepresentation {
```

Comment:
It's upsetting that we need to make this `pub` because it's really a backwards compatibility artifact specifically for `automerge-wasm`. I can't think of a neat way to avoid it though. On that basis I think we should do two things:

1. Add some documentation to the `Array` variant explaining that new implementations should not use this representation.
2. Given that we have this `TextRepresentation` enum in our public API we can at least replace `OpObserver::test_as_seq` with `OpObserver::text_representation() -> TextRepresentation` which will be clearer